core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java

```diff
@@ -43,6 +48,40 @@ public ErrorCollector(Config config) {
     this.config = config;
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+
+    Schema inputSchema = context.getInputSchema();
+    if (inputSchema == null || inputSchema.getFields() == null) {
+     return;
+    }
+
+    List<FieldOperation> fieldOperations = new ArrayList<>();
+    // the originial fields will not change in transform
+    inputSchema.getFields().forEach(
+      field -> fieldOperations.add(new FieldTransformOperation(String.format("Error collector for field %s",
+                                                                             field.getName()),
+                                                              "Error collector identity transform",
+                                                               Collections.singletonList(field.getName()),
+                                                               field.getName())));
+
+    // emit records for generated fields
+    if (config.messageField != null) {
```

Comment:
What if input schema does not have this field? I think we should also check `inputSchema.getField(config.messageField) != null`
Also we should add empty string check

Reply:
The input schema is guaranteed to not have this field, we did the validation check in `configurePipeline()`, this field is a generated field in the plugin, that's why the fll operation for this field does not contain the input fields.